libImaging/Quant.c

```diff
@@ -1251,13 +1271,20 @@ quantize(Pixel *pixelData,
    free_box_tree(root);
    root=NULL;
 
-   qp=malloc(sizeof(uint32_t)*nPixels);
+   /* malloc check ok, using calloc for overflow */
+   qp=calloc(nPixels, sizeof(uint32_t));
    if (!qp) { goto error_4; }
 
-   avgDist=malloc(sizeof(uint32_t)*nPaletteEntries*nPaletteEntries);
+   if ((nPaletteEntries > UINT32_MAX / nPaletteEntries ) ||
+       (nPaletteEntries > UINT32_MAX / (nPaletteEntries * sizeof(uint32_t)))) {
```

Comment:
I think first, you need to check `nPaletteEntries > UINT32_MAX / sizeof(uint32_t)`.
The last check is performed by calloc.